src/main/kotlin/dev/pfilaretov42/aoc/aoc2023/day4/Scratchcards.kt

```diff
@@ -5,8 +5,37 @@ import kotlin.math.pow
 
 fun main() {
     val inputFile = File("input/2023/day4/input")
-    partOne(inputFile)
-//    partTwo(inputFile)
+//    partOne(inputFile)
+    partTwo(inputFile)
+}
+
+fun partTwo(inputFile: File) {
+    val lineRegex = ":\\s+".toRegex()
+    val cardRegex = "\\s+\\|\\s+".toRegex()
+    val numbersRegex = "\\s+".toRegex()
+
+    val cardsCountMap = mutableMapOf<Int, Int>()
+
+    inputFile.forEachLine { line ->
+        val cardAndNumbers = line.split(lineRegex)
+
+        val cardNumber = cardAndNumbers[0].split(numbersRegex)[1].toInt()
+        val cardsCount = cardsCountMap.merge(cardNumber, 1) { oldValue, _ -> oldValue + 1 }!!
+
+        val numbers = cardAndNumbers[1].split(cardRegex)
+        val winningNumbers = numbers[0].split(numbersRegex).toSet()
+        val numbersToTest = numbers[1].split(numbersRegex).toSet()
+
+        val winningCount = numbersToTest.filter { winningNumbers.contains(it) }.size
+
+        IntRange(cardNumber + 1, cardNumber + winningCount).forEach { additionalCardNumber ->
+            cardsCountMap.merge(additionalCardNumber, 1) { oldValue, _ -> oldValue + cardsCount }
+        }
+
+        println(cardsCountMap.values.sum())
+    }
+
+    println(cardsCountMap.values.sum())
```

Comment:
**Suggestion:** The `println(cardsCountMap.values.sum())` statement is repeated twice, once inside the loop and once outside. If you only need the final sum, you can remove the one inside the loop. [enhancement]
```suggestion
        println(cardsCountMap.values.sum())
```